content/shmem_init_thread.tex

```diff
@@ -0,0 +1,55 @@
+\apisummary{
+Initializes the OpenSHMEM library, similar to \FUNC{shmem\_init}, and performs any
+initialization required for supporting the provided thread level.}
+
+%Initializes the \openshmem{} library, similar to \FUNC{shmem\_init}, and in addition performs the initialization required for thread-safe invocation of \openshmem{} functions.}
+
+\begin{apidefinition}
+
+\begin{Csynopsis}
+int shmem_init_thread(int requested, int *provided);
+\end{Csynopsis}
+
+\begin{apiarguments}
+\apiargument{IN}{requested}  {The thread level support requested by the user.
+}
+\apiargument{OUT}{provided}{The thread level support provided by the \openshmem{} implementation.}
+\end{apiarguments}
+
+
+\apidescription{
+\FUNC{shmem\_init\_thread}  initializes the \openshmem{} library similar to
+\FUNC{shmem\_init}, and in addition performs the initialization required for
+supporting the provided thread level. The argument
+\VAR{requested} is used to specify the desired level of thread support.
+The argument \VAR{provided} returns the support level provided by the library.
+The correct values for \VAR{provided} and \VAR{requested} are 
+\CONST{SHMEM\_THREAD\_SINGLE}, \newline 
+\CONST{SHMEM\_THREAD\_FUNNELED}, \CONST{SHMEM\_THREAD\_SERIALIZED}, or 
+\CONST{SHMEM\_THREAD\_MULTIPLE}.
+     
+An \openshmem{} program is initialized either by \FUNC{shmem\_init} or
+\FUNC{shmem\_init\_thread}. Similar to \FUNC{shmem\_init}, the
+\FUNC{shmem\_init\_thread} may not be called multiple times in an \openshmem{} program.
+ If the call to \FUNC{shmem\_init\_thread} is unsuccessful in allocating and initializing
+ resources for the \openshmem{} library, then the behavior of any subsequent call
+ to the \openshmem{} library is undefined.
+}
+
+\apireturnvalues{    
+	  \FUNC{shmem\_init\_thread} returns 0 upon success; otherwise, it returns a non-zero
+      value.
+    }
+
+\apinotes{
+    The \openshmem{} library can be initialized either by \FUNC{shmem\_init} or
+    \FUNC{shmem\_init\_thread}. If the \openshmem{} library  is initialized by
+    \FUNC{shmem\_init}, the library implementation can choose to support one of the 
+    defined thread levels.
```

Comment:
Updated rewording suggestion: If the \openshmem library is initialized by \FUNC{shmem_init}, the provided thread level is implementation-defined.  However, the provided thread level must conform to one of the threading levels defined in Section~\ref{sec:thread-info}.